packages/synapse-react-client/src/components/SynapseTable/datasets/DatasetItemsEditor.test.tsx

```diff
@@ -136,22 +136,24 @@ async function clickRemove() {
 
 const mockToastFn = displayToast
 
-const mockOnUnsavedChangesFn = jest.fn()
-const mockOnSaveFn = jest.fn()
-const mockOnCloseFn = jest.fn()
-
 // Captures the JSON passed to the server via msw.
 const updatedEntityCaptor = jest.fn()
 
-const defaultProps: DatasetItemsEditorProps = {
-  entityId: mockDatasetEntityData.id,
-  onSave: mockOnSaveFn,
-  onClose: mockOnCloseFn,
-  onUnsavedChangesChange: mockOnUnsavedChangesFn,
-}
-
 async function renderComponent(wrapperProps?: SynapseContextType) {
-  const wrapper = render(<DatasetItemsEditor {...defaultProps} />, {
+  const user = userEvent.setup()
+
+  const mockOnUnsavedChangesFn = jest.fn()
+  const mockOnSaveFn = jest.fn()
+  const mockOnCloseFn = jest.fn()
```

Comment:
Fixed some flaky test issues by creating new mocks/`userEvent` instance each render. I think I had an issue where some actions in tests would fire after the test had ended and next test started, which made mock invocation counts incorrect.